flint/flint-spark-integration/src/main/scala/org/apache/spark/sql/flint/datatype/FlintDataType.scala

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.apache.spark.sql.flint.datatype
+
+import org.json4s.{Formats, JField, JValue, NoTypeHints}
+import org.json4s.JsonAST.{JNothing, JObject, JString}
+import org.json4s.jackson.JsonMethods
+import org.json4s.native.Serialization
+
+import org.apache.spark.sql.types._
+
+/**
+ * Mapping between Flint data type and Spark data type
+ */
+object FlintDataType {
+
+  implicit val formats: Formats = Serialization.formats(NoTypeHints)
+
+  /**
+   * parse Flint metadata and extract properties to StructType.
+   */
+  def deserialize(metadata: String): StructType = {
+    deserializeJValue(JsonMethods.parse(metadata))
+  }
+
+  def deserializeJValue(json: JValue): StructType = {
+    val properties = (json \ "properties").extract[Map[String, JValue]]
+    val fields = properties.map { case (fieldName, fieldProperties) =>
+      deserializeFiled(fieldName, fieldProperties)
+    }
+
+    StructType(fields.toSeq)
+  }
+
+  def deserializeFiled(fieldName: String, fieldProperties: JValue): StructField = {
+    val metadataBuilder = new MetadataBuilder()
+    val dataType = fieldProperties \ "type" match {
+      // boolean
+      case JString("boolean") => BooleanType
```

Comment:
Could you update the field type section here: https://github.com/opensearch-project/sql/blob/feature/flint/flint/docs/index.md#field-data-type ? Maybe add a table as SQL for mapping from Spark SQL Type -> Flint Data Type -> OpenSearch Field Type (the last two are the same for now)

Reply:
sure. will add datatype section and mapping in the following PR.